File: packages/rust/examples/basic.rs
#![allow(missing_docs)]

use erpc_sdk::{ErpcClient, ErpcClientConfig};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let api_key = std::env::var("ERPC_API_KEY")?;
    let erpc = ErpcClient::new(ErpcClientConfig::new(api_key))?;

    let slot = erpc
        .solana
        .rpc
        .get_slot(Vec::<serde_json::Value>::new())?
        .send()
        .await?;
    let chain_id = erpc.ethereum.rpc.eth_chain_id().send().await?;

    println!("slot={slot}, chain_id={chain_id}");
    erpc.close().await;
    Ok(())
}

File: packages/ruby/examples/basic.rb
# frozen_string_literal: true

require "erpc_sdk"

begin
  erpc = ERPC::Client.new(
    ERPC::ClientConfig.new(api_key: ENV.fetch("ERPC_API_KEY"))
  )

  slot = erpc.solana.rpc.get_slot.send
  chain_id = erpc.ethereum.rpc.eth_chain_id.send
  avalanche_chain_id = erpc.avalanche.rpc.eth_chain_id.send
  puts({ slot: slot, chain_id: chain_id, avalanche_chain_id: avalanche_chain_id })
ensure
  erpc&.close
end
